### PR TITLE
Implement adjoints for atleast_*d

### PR DESCRIPTION
**Pull request by dmitriy-serdyuk**

### CODE REVIEW

**Comment by dmitriy-serdyuk** [comment]
I signed it!

**Comment by alexbw** [comment]
Awesome, thanks! Could you please add some tests as well? You can [see the guide to adding new adjoints here](https://github.com/google/tangent/blob/master/CONTRIBUTING.md#adding-new-derivatives). It would be nice to add the forward-mode derivatives as well, but that is not strictly necessary.

**Comment by dmitriy-serdyuk** [comment]
Done, it passes locally. Should I name inputs as `x`?

BTW, how do you implement `d` object?

**Comment by dmitriy-serdyuk** [comment]
Concerning forward derivatives, they are already implemented as far as I understand https://github.com/google/tangent/blob/master/tangent/tangents.py#L206-L218.

File: tests/functions.py [hunk 1]
```diff
@@ -435,6 +435,18 @@ def numpy_arctan(a):
...
+def numpy_atleast_1d(a):
+  return np.atleast_1d(a)
+
+
+def numpy_atleast_2d(a):
+  return np.atleast_2d(a)
+
+
+def numpy_atleast_3d(a):
+  return np.atleast_3d(a)
```
Comment (alexbw) [inline tests/functions.py]: If the input variable to a test function is `x`, we pass in an array. I think this is what you want to do. Then, the output should probably be a scalar, so return `np.sum(np.atleast_*d(x))`

**Comment by alexbw** [comment]
Oh, nice re: forward-mode derivatives. Also, please see my comments re: the specific form of the test -- we want to test vector-valued inputs.

`d[]` is a little code macro that tangent internally will search for, and replace with the correct name of the derivative of the variable name enclosed. This is _usually_ mechanical: `d[x]` is replaced with `bx` in reverse mode and `dx` in forward mode. However, if you've already used the variable `bx` in your original code, we have to find a unique name. Tangent takes care of all of this internally in [the `replace` function](https://github.com/google/tangent/blob/e3be5784a0850dbc85addcfdd233d4af181c7e6d/tangent/template.py#L144-L170).

File: tests/functions.py (hunk 1 above)
Comment (alexbw) [reply tests/functions.py]: Although if the tests pass without wrapping it with `np.sum`, that is fine, too.

**Comment by mdanatg** [comment]
Thanks for tuning the tests! I think that addresses the requests Alex had, so I'll go ahead and pull the branch.

**Pull request merged**